Documentation/kubernetes-networking.md

```diff
@@ -0,0 +1,91 @@
+# Kubernetes Networking
+
+## Network Model
+
+The Kubernetes network model outlines four methods of component communication:
+
+* Container-to-Container Communication
+    * Containers launched within the same pod can all reach each other's ports via localhost. This simplifies / secures communication between applications launched in the same Pod.
+
+* Pod-to-Pod Communication
+    * Each Pod in a Kubernetes cluster is assigned an IP in a flat shared networking namespace. This allows for a clean network model where Pods, from a networking perspective, can be treated much like VMs or physical hosts.
+
+* Pod-to-Service Communication
+    * Services are implemented by assigning Virtual IPs which clients can access and are transparently proxied to the Pods grouped by that service. Requests to the Service IPs are intercepted by a kube-proxy process running on all nodes, which is then responsible for routing to the correct Pod.
+
+* External-to-Internal Communication
+    * Accessing services from outside the cluster is generally implemented by configuring external loadbalancers which target all nodes in the cluster. Once traffic arrives at a node, it can be routed to the correct Service backends via the kube-proxy.
+
+See [Kubernetes Networking][kubernetes-network] for more detailed information on the Kubernetes network model and motiviation.
+
+[kubernetes-network]: http://kubernetes.io/v1.0/docs/admin/networking.htm
+
+## Port allocation
+
+The information below describes a minimum set of port allocations used by Kubernetes components. Some of these allocations will be optional depending on the deployment (e.g. if flannel is being used). Additionally, there is likely additional ports a deployer will need to open on their infrastructure (e.g. 22/ssh).
+
+Controller Node Inbound
+
+| Protocol | Port Range | Source                       | Purpose                |
+-----------|------------|------------------------------|------------------------|
+| TCP      | 443        | Worker Nodes & API Consumers | Kubernetes API server. |
+
+Worker Node Inbound
+
+| Protocol | Port Range  | Source                         | Purpose                                                                |
+-----------|-------------|--------------------------------|------------------------------------------------------------------------|
+| TCP      | 10250       | Control Nodes                  | Worker node Kubelet healthcheck port.                                  |
+| TCP      | 30000-32767 | ALL or External Load-balancers | Default port range for [external service][external-service] node ports |
```

Comment:
The external port range does not necessarily need to be exposed to the outside world. Typically you would expose it to your load-balancers which are exposed publicly. But this isn't strictly necessary. Maybe I could try and explain this in the purpose field instead of source.